airflow/models/dag.py

```diff
@@ -1506,6 +1505,17 @@ def safe_dag_id(self):
         return self.dag_id.replace('.', '__dot__')
 
     def get_dag(self):
+        # TODO: Resolve this in upstream by storing relative path in db (config driven)
+        try:
+            # fix for manual trigger, as db stores scheduler path which is different from webserver
+            # path, as a result, this function returns NONE
+            path_regex = "airflow_scheduler-.-[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[" \
```

Comment:
Wouldn't this break for locally run airflow clusters?

Can we decouple this from the airflow scheduler job name? Currently if we change the job name it will cause a very hard to debug bug, and will require an Airflow release to change the path here.

I feel like it's probably less brittle, simpler, and less likely to break to just take the last couple of sections of path.split(os.path.sep) rather than have this regex logic. You can hardcode dags_folder in server role configs since it shouldn't even be configurable, and your change would break on a custom dags_folder anyways, and then we can add a tiny internal unit test that will basically just compare the constant with the hardcoded dag folder with a specific string with an error message like "Airflow code <link to airflow code> assumes dag folder looks like X/Y/Z, if you want to change the format you need to update the Airflow code as well and then change this test to match>. This will prevent someone from accidentally breaking Airflow internally in a hard to debug way in the future if they change the dags_folder.

Reply:
This won't break in local, as if the path doesn't match the regex, it picks the default path.

This is what I am planning to do. If we can store the relative path in db and make it config driven, then we can resolve this.
So, path in db would be stored as "DAG_HOME_PATH/<relative_path>".
where 'DAG_HOME_PATH' would be a string, which would be replaced by $DAG_HOME_PATH in respective webserver, scheduler, and worker.

Reply:
If we are going to do that solution for open source first then I'm ok with it. Otherwise I don't like the idea of storing things in the DB, the DB is fork that's harder to manage than code, we can put it the airflow config instead.

If you are planning on making the change in open source I'm not sure how easy that would be to do, Sumit might have more context and should be able to help you out there.

Reply:
Have seen some issues in open-source already due to full path being stored in DB, so it'll be a great improvement overall to store the relative path in DB instead of a full path. Also, we are not talking about a fork here and will try to get this done in open-source only, but till that time we've to rely on some brittle solution like this (regex).